test/Plugin/TestAsset/JsonSerializableEntity.php

```diff
@@ -10,10 +10,10 @@
 class JsonSerializableEntity extends Entity implements JsonSerializable
 {
     /**
-     * @return array
+     * @return array<string,string>
      */
     #[ReturnTypeWillChange]
-    public function jsonSerialize()
+    public function jsonSerialize(): mixed
```

Comment:
`: array` and drop the `ReturnTypeWillChange` ?

Reply:
Thank you @gsteel 

Since PHP 8.1, the [JsonSerializable  interface](https://www.php.net/manual/en/jsonserializable.jsonserialize.php) is set as `jsonSerialize(): mixed`, this is why I had to add it and override it on the DocBlock.

This change should make it compatible with all PHP 8 versions and avoid the `Declaration of FooClass::jsonSerialize() should be compatible with JsonSerializable::jsonSerialize(): mixed` meessage

Reply:
Return types are covariant, so `array` is groovy: https://3v4l.org/gnRkO